addons/common/fnc_randomizeFacewear.sqf

```diff
@@ -0,0 +1,43 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_randomizeFacewear
+
+Description:
+    Add config defined weighted random facewear to unit.
+    Uses same config as headgearList[].
+
+Parameters:
+    _unit - unit <OBJECT>
+
+Returns:
+    true on success, false on error <BOOLEAN>
+
+Examples:
+    (begin example)
+        [unit] call CBA_fnc_randomizeFacewear;
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+
+params [["_unit", objNull]];
+
+if (isNull _unit) exitWith {
+    TRACE_1("unit is null",_unit);
+    false
+};
+
+if (!local _unit || {!(_unit getVariable ["BIS_enableRandomization", true])}) exitWith {true};
+
+private _allowedFacewear = getArray (configFile >> "CfgVehicles" >> typeOf _unit >> "CBA_allowedFacewear");
+
+if (_allowedFacewear isEqualTo []) exitWith {true};
+
+private _facewear = selectRandomWeighted _allowedFacewear;
+
+if (_facewear == "") then {
+    removeGoggles _unit;
+} else {
+    _unit addGoggles _facewear;
+};
```

Comment:
Header says returns true on success and false on error. This never returns false though and may return nil.
What would be considered failure? Disabled randomization by class from mission config or by object from object namespace variable?

Reply:
It returns `false` if passed unit is null. Forgot to add a `true` at the end. FYI, I wrote this PR on my phone. :eyes: 

I am not sure.

Reply:
Okay, that is acceptable.
```suggestion

true

```